src/lib.rs

```diff
@@ -258,9 +258,9 @@ psm_stack_manipulation! {
 
         fn page_size() -> usize {
             // FIXME: consider caching the page size.
-            #[cfg(not(target_arch = "wasm32"))]
+            #[cfg(not(any(target_arch = "wasm32",target_os = "hermit")))]
             unsafe { libc::sysconf(libc::_SC_PAGE_SIZE) as usize }
-            #[cfg(target_arch = "wasm32")]
+            #[cfg(any(target_arch = "wasm32",target_os = "hermit"))]
```

Comment:
If there is absolutely no way to obtain the page size on hermit then lets hardcode it to something more appropriate for platforms hermit runs on (probably 4kB.)